ea_airflow_util/providers/sharefile/hooks/sharefile.py

```diff
@@ -76,6 +78,43 @@ def download(self, item_id, local_path):
             self.log.error('Item request failed with status code: {}'.format(dl_response.status_code))
             dl_response.raise_for_status()
 
+    def upload_file(self, folder_id, local_file):
+        # establish a session if we don't already have one
+        if not self.session:
+            self.get_conn()
+
+        # referencing sharefile's upload logic here
+        # https://api.sharefile.com/samples/python
+        upload_config_uri = f"{self.base_url}/Items({folder_id})/Upload"
+        upload_config_resp = self.session.get(upload_config_uri)
+        upload_config = upload_config_resp.json()
+    
+        upload_uri = ""
+        try:
+            upload_uri = upload_config["ChunkUri"]
+        except KeyError:
+            # most likely an invalid or nonexistent folder ID
+            raise AirflowException(f"Failed to get upload link for {local_file}; Reason: {upload_config['reason']}; Message: {upload_config['message']['value']}")
+
+        files = {Path(local_file).name: open(local_file, 'rb')}
+        resp = self.session.post(upload_uri, files=files)
+        resp.raise_for_status()
+
+
+    def folder_id_from_path(self, folder_path):
+        """Given a complete Sharefile folder path, returns the ID of the leaf folder. If the path is not found, returns `None`"""
+        folders = self.find_folders("allshared")
```

Comment:
Tell me more about the "allshared" string passed to find_folders.

Reply:
So I think my point of confusion may relate to a misunderstanding of the ShareFile API. Is "allshared" a built-in keyword, or does it mean something specific to the API or the `find_folders()` method?

Reply:
It's a way to tell the Sharefile API to return all resources underneath the root. I tried to explain this with my comment but let me know if I can clarify it

Reply:
Yeah, I think a description that "allshared" is something Sharefile-specific would be helpful. I didn't realize ShareFile had hard-coded that string as a folder ID. What happens if we create a folder called "allshared"?

Reply:
I took another pass at it